server/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -312,15 +318,34 @@ public static Item forUpdate(String id,
                 seqNo,
                 primaryTerm,
                 List.of(),
-                Translog.UNSET_AUTO_GENERATED_TIMESTAMP
+                Translog.UNSET_AUTO_GENERATED_TIMESTAMP,
+                usedBytes
             );
         }
 
         public static Item forInsert(String id,
                                      List<String> pkValues,
                                      long autoGeneratedTimestamp,
+                                     @Nullable Reference[] insertColumns,
                                      @Nullable Object[] values,
                                      @Nullable Symbol[] onConflictAssignments) {
+            long usedBytes = SHALLOW_SIZE;
```

Comment:
I tried first a variant where `insertColumns` is passed to `ramBytesUsed` but then we can't implement the `Accountable` interface and in some cases we don't have access to the `insertColumns`. For example in:

https://github.com/crate/crate/blob/50585246e8b1ab83f26995960b5e3d5192e8355c/server/src/main/java/io/crate/execution/engine/indexing/ShardedRequests.java#L61